specification/attestation/Attestation/models.tsp

```diff
@@ -0,0 +1,926 @@
+
+using Azure.ClientGenerator.Core;
+
+namespace Microsoft.Attestation;
+
+/**
+ * Content type for upload
+ */
+#suppress "@azure-tools/typespec-azure-core/no-closed-literal-union" "Intentionally closed for HTTP Content-Type header compliance"
+union SigningCertificatesContentType {
+  /**
+   * Content Type 'application/jwk+json'
+   */
+  "application/jwk+json",
+
+  /**
+   * Content Type 'application/json'
+   */
+  "application/json",
+}
+
+/**
+ * AttestationType
+ */
+union AttestationType {
+  string,
+
+  /**
+   * Intel Software Guard eXtensions
+   */
+  SgxEnclave: "SgxEnclave",
+
+  /**
+   * OpenEnclave extensions to SGX
+   */
+  OpenEnclave: "OpenEnclave",
+
+  /**
+   * Azure Guest Attestation
+   */
+  AzureGuest: "AzureGuest",
+
+  /**
+   * AMD SEV SNP Virtual Machine
+   */
+  SevSnpVm: "SevSnpVm",
+
+  /**
+   * Edge TPM Virtualization Based Security
+   */
+  Tpm: "Tpm",
+
+  /**
+   * Intel TDX Virtual Machine
+   */
+  TdxVm: "TdxVm",
+}
+
+/**
+ * Specifies the type of the data encoded contained within the "data" field of a
+ * "RuntimeData" or "InitTimeData" object
+ */
+union DataType {
+  string,
+
+  /**
+   * The field's content should be treated as binary and not interpreted by MAA.
+   */
+  Binary: "Binary",
+
+  /**
+   * The field's content should be treated as UTF-8 JSON text that may be further
+   * interpreted by MAA. Refer to RFC 8259 for a description of JSON serialization
+   * standards for interoperability.
+   */
+  JSON: "JSON",
+}
+
+/**
+ * The result of the operation
+ */
+@usage(Usage.output)
+union CertificateModification {
+  string,
+
+  /**
+   * After the operation was performed, the certificate is in the set of
+   * certificates.
+   */
+  IsPresent: "IsPresent",
+
+  /**
+   * After the operation was performed, the certificate is no longer present in the
+   * set of certificates.
+   */
+  IsAbsent: "IsAbsent",
+}
+
+/**
+ * The result of the operation
+ */
+@usage(Usage.output)
+union PolicyModification {
+  string,
+
+  /**
+   * The specified policy object was updated.
+   */
+  Updated: "Updated",
+
+  /**
+   * The specified policy object was removed.
+   */
+  Removed: "Removed",
+}
+
+/**
+ * The response to an attestation policy operation
+ */
+model PolicyResponse {
+  /**
+   * An RFC7519 JSON Web Token structure whose body is an PolicyResult object.
+   */
+  @pattern("[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]*\\.[A-Za-z0-9_-]*")
+  token?: string;
+}
+
+/**
+ * An error response from Attestation.
+ */
+@error
+model CloudError {
+  /**
+   * An error response from Attestation.
+   */
+  error?: CloudErrorBody;
+}
+
+/**
+ * An error response from Attestation.
+ */
+model CloudErrorBody {
+  /**
+   * An identifier for the error. Codes are invariant and are intended to be
+   * consumed programmatically.
+   */
+  code?: string;
+
+  /**
+   * A message describing the error, intended to be suitable for displaying in a
+   * user interface.
+   */
+  message?: string;
+}
```

Comment:
This looks similar enough that you might be able to use the error type in the azure.core